scipy/stats/_stats_py.py

```diff
@@ -316,21 +307,14 @@ def gmean(a, axis=0, dtype=None, weights=None):
     2.80668351922014
 
     """
-    if not isinstance(a, np.ndarray):
-        # if not an ndarray object attempt to convert it
-        a = np.array(a, dtype=dtype)
-    elif dtype:
```

Comment:
Even if `not dtype`, `asarray` will just preserve the input type. Why branch when we don't need to?